src/io/parser.cpp

```diff
@@ -4,6 +4,7 @@
  */
 #include "parser.hpp"
 
+#include <algorithm>
```

Comment:
Please stick to the Google style guide and move this after `<string>`:
https://github.com/microsoft/LightGBM/pull/2066#issue-263709049
As an example:

https://github.com/microsoft/LightGBM/blob/3e3d76f64bfa3d95f9769bd82be7124158ff3df7/src/objective/binary_objective.hpp#L11-L12

Reply:
ah I see, will do!

Reply:
Thanks!